src/link.js

```diff
@@ -34,3 +34,150 @@ export default class Link extends Plugin {
+ * quirky results. The same applies if multiple manual or automatic decorators were defined for the same attribute.
+ *
+ * **Note**: Since the `target` attribute management for external links is a common use case, there is a predefined automatic decorator
+ * dedicated for that purpose which can be enabled by turning a single option on. Check out the
+ * {@link module:link/link~LinkConfig#targetDecorator `config.link.targetDecorator`} configuration description to learn more.
+ *
+ * @member {Array.<module:link/link~LinkDecoratorAutomaticOption|module:link/link~LinkDecoratorManualOption>}
+ * module:link/link~LinkConfig#decorators
+ */
+
+/**
+ * Describes an automatic link {@link module:link/link~LinkConfig#decorators decorator}. This kind of a decorator matches
+ * all links in the editor content against a function which decides whether the link should gain a pre–defined set of attributes
+ * or not.
+ *
+ * It takes an object with key-value pairs of attributes and a callback function which must return a boolean based on link's
+ * `href` (URL). When the callback returns `true`, attributes are applied to the link.
+ *
+ * For example, to add the `target="_blank"` attribute to all links in the editor starting with the `http://`,
+ * then configuration could look like this:
+ *
+ *		{
+ *			mode: 'automatic',
+ *			callback: url => url.startsWith( 'http://' ),
+ *			attributes: {
+ *				target: '_blank'
+ *			}
+ *		}
+ *
+ * **Note**: Since the `target` attribute management for external links is a common use case, there is a predefined automatic decorator
+ * dedicated for that purpose which can be enabled by turning a single option on. Check out the
+ * {@link module:link/link~LinkConfig#targetDecorator `config.link.targetDecorator`} configuration description to learn more.
+ *
+ * @typedef {Object} module:link/link~LinkDecoratorAutomaticOption
+ * @property {'automatic'} mode The kind of the decorator. `'automatic'` for all automatic decorators.
+ * @property {Function} callback Takes an `url` as a parameter and returns `true` if the `attributes` should be applied to the link.
+ * @property {Object} attributes Key-value pairs used as link attributes added to the output during the
+ * {@glink framework/guides/architecture/editing-engine#conversion downcasting}.
+ * Attributes should follow the {@link module:engine/view/elementdefinition~ElementDefinition} syntax.
+ */
+
+/**
+ * Describes a manual link {@link module:link/link~LinkConfig#decorators decorator}. This kind of a decorator is represented in
+ * the link feature's {@link module:link/linkui user interface} as a switch the user can use to control the presence
+ * of a pre–defined set of attributes.
+ *
+ * For instance, to allow users to manually control the presence of the `target="_blank"` and
+ * `rel="noopener noreferrer"` attributes on specific links, the decorator could look as follows:
+ *
+ *		{
+ *			mode: 'manual',
+ *			label: 'Open link in new window',
```

Comment:
We need to document how to translate those labels. In the [heading configuration](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_heading_heading-HeadingConfig.html#member-options) you can specify some standard strings which are [automatically translated](https://github.com/ckeditor/ckeditor5-heading/blob/master/src/utils.js#L17). Same for [alignment](https://github.com/ckeditor/ckeditor5-alignment/blob/master/src/alignmentui.js#L51), font color, etc..

I think the same should happen here at least for `'Open link in new tab'` and developers must know about it.

Reply:
@oleq If I understand correctly `t()` method accepts **only** strings, at least this is how I understand how it works after talk with PK.
So this case is completely different case than Heading, Font, Alignment, etc. We don't have strings for those labels, because those are defined by the developer, so we cannot provide any translations for them.
Originally there was translations for the labels, but I removed them when I noticed this situation and errors during snippet building.
https://github.com/ckeditor/ckeditor5-link/pull/222/commits/83cf302239663a7384529f79d75f9fc76d5c2fd6

If I understand this incorrectly, please give me more info how to approach to translation for unknown strings.

Reply:
Generally speaking, you're right. Developers define those strings in the editor configuration, when `t()` is unavailable. 

But just like in the heading feature, we can hardcode certain translations for the most popular options and tell developers that they can use those if they want them to be translated.

In the heading feature, it's developers who define [`config.heading.options`](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_heading_heading-HeadingConfig.html#member-options). They can customize them to their liking but as long as they use `'Paragraph'` or `'Heading 3'` in `Title`, those strings will translate to other languages like French or German.

We can do the same for the `'Open in a new tab'` toggle switch option. If enabled as a manual decorator, regardless of the decorator configuration (e.g. with `rel` or without), the thing will always translate. Same with `'Downloadable'`, which is another label people are gonna use.